java/internal/src/cuvs_java.c

```diff
@@ -91,3 +93,33 @@ void search_cagra_index(cuvsCagraIndex_t index, float *queries, int topk, long n
   cudaMemcpy(neighbors_h, neighbors, sizeof(uint32_t) * n_queries * topk, cudaMemcpyDefault);
   cudaMemcpy(distances_h, distances, sizeof(float) * n_queries * topk, cudaMemcpyDefault);
 }
+
+typedef struct {
+    char name[64];
+    size_t totalMemory;
+    size_t freeMemory;
+} GpuDetail;
+
+int get_gpu_details(GpuDetail *details, int maxGpus) {
+    int deviceCount = 0;
+    cudaError_t err = cudaGetDeviceCount(&deviceCount);
+
+    if (err != cudaSuccess || deviceCount == 0) {
+        return -1;
+    }
+    else if(deviceCount == 0){
+        return 0;
+    }
```

Comment:
Why is this else block even needed? We are eventually returning `deviceCount, ' right?

Reply:
OK will just return num of GPUs and if error, return -1.